### ISSUE TITLE
Android App Crashes as soon as it hits BackgroundGeolocation.start()

### ISSUE THREAD

**Issue by onkar0777** [mauron85/cordova-plugin-background-geolocation#404]
## Your Environment

* Plugin version: 3.0.0-alpha.14 (tried with 2.3.5 as well)
* Platform: Android
* OS version: 6.0.1
* Device manufacturer and model: OnePlus 2
* Cordova version (`cordova -v`): 8.0.0
* Cordova platform version (`cordova platform ls`): 7.0.0
* Plugin configuration options: BackgroundGeolocation.configure({
            locationProvider: BackgroundGeolocation.DISTANCE_FILTER_PROVIDER, (Tried with others also)
            desiredAccuracy: BackgroundGeolocation.HIGH_ACCURACY,
            stationaryRadius: 50,
            distanceFilter: 50,
            notificationTitle: 'Background tracking',
            notificationText: 'Enabled for Vehicle Tracking',
            debug: true,
            interval: 10000,
            fastestInterval: 5000,
            activitiesInterval: 10000,
            url: serverLocationUrl+'?routeId='+routeId,
            httpHeaders: {
                'authorization': $window.UTILITIES.LOCAL_STORAGE.get('auth')
            },
            // customize post properties
            postTemplate: {
                lat: '@latitude',
                lon: '@longitude',
                bear: '@bearing'
            }
        });
* Link to your project:

## Context

As soon as the code reaches backgroundlocation.start() , the app crashes

## Expected Behavior

should run and not crash

## Actual Behavior

app crashed

## Steps to Reproduce

1. The given example does not work for me, followed the same steps

## Debug logs

04-13 18:38:22.412  4260   775 I Authzen : [DeviceStateSyncManager] The server is in sync with current state. Nothing to do
04-13 18:38:22.629  1124  1124 I DeviceIdleController: updateChargingLocked: charging=true
04-13 18:38:22.629  1124  1124 I DeviceIdleController: becomeActiveLocked, reason = charging
04-13 18:38:22.636 11583 11583 D AbstractProcessor: onBatteryChange Percent:76 State:2 
04-13 18:38:22.784  3800  4078 D NetworkController.MobileSignalController(3): onSignalStrengthsChanged signalStrength=SignalStrength: 99 0 -120 -160 -120 -1 -1 16 -111 -10 80 2147483647 2147483647 2 2 gsm|lte level=2 voicelevel=2 datalevel=2
04-13 18:38:22.834  7402  7402 W Settings: Setting airplane_mode_on has moved from android.provider.Settings.System to android.provider.Settings.Global, returning read-only value.
04-13 18:38:23.039  7402  7402 W Settings: Setting preferred_network_mode has moved from android.provider.Settings.Secure to android.provider.Settings.Global.
04-13 18:38:23.457  4063  4063 D wpa_supplicant: wlan0: Control interface command 'SIGNAL_POLL'
04-13 18:38:23.462  4063  4063 D wpa_supplicant: CTRL-DEBUG: global_ctrl_sock-sendto: sock=11 sndbuf=212992 outq=0 send_len=48
04-13 18:38:24.461   352   352 W auditd  : type=1400 "ipacm""/dev/socket/ipacm_log_file"
04-13 18:38:24.461   352   352 W auditd  : type=1400 "ipacm""/dev/socket/ipacm_log_file"
04-13 18:38:24.461   352   352 W auditd  : type=1400 "ipacm""/dev/socket/ipacm_log_file"
04-13 18:38:24.462   352   352 W auditd  : type=1400 "ipacm""/dev/socket/ipacm_log_file"
04-13 18:38:24.549  7402  7402 W Settings: Setting preferred_network_mode has moved from android.provider.Settings.Secure to android.provider.Settings.Global.
04-13 18:38:24.550  7402  7402 W Settings: Setting preferred_network_mode has moved from android.provider.Settings.Secure to android.provider.Settings.Global.
04-13 18:38:24.551  7402  7402 W Settings: Setting preferred_network_mode has moved from android.provider.Settings.Secure to android.provider.Settings.Global.
04-13 18:38:25.035 29487 29487 D SystemWebChromeClient: file:///android_asset/www/js/angular/controllers/map-controller.js: Line 10 : 5ad08553a5172c2e88b2e8e3
04-13 18:38:25.035 29487 29487 I chromium: [INFO:CONSOLE(10)] "5ad08553a5172c2e88b2e8e3", source: file:///android_asset/www/js/angular/controllers/map-controller.js (10)
04-13 18:38:25.036 29487 29487 D SystemWebChromeClient: file:///android_asset/www/js/angular/controllers/bus-routes-controller.js: Line 61 : app.map
04-13 18:38:25.036 29487 29487 I chromium: [INFO:CONSOLE(61)] "app.map", source: file:///android_asset/www/js/angular/controllers/bus-routes-controller.js (61)
04-13 18:38:25.036 29487 29487 D SystemWebChromeClient: file:///android_asset/www/js/angular/controllers/bus-routes-controller.js: Line 62 : app.busroutes
04-13 18:38:25.036 29487 29487 I chromium: [INFO:CONSOLE(62)] "app.busroutes", source: file:///android_asset/www/js/angular/controllers/bus-routes-controller.js (62)
04-13 18:38:25.063 29487   811 D com.marianhello.bgloc.data.sqlite.SQLiteConfigurationDAO: Configuration persisted with rowId = 1
04-13 18:38:25.064 29487   811 D com.marianhello.bgloc.BackgroundGeolocationFacade: Service configured with: Config[distanceFilter=50 stationaryRadius=50.0 desiredAccuracy=0 interval=10000 fastestInterval=5000 activitiesInterval=10000 isDebugging=true stopOnTerminate=true stopOnStillActivity=true startOnBoot=false startForeground=true locationProvider=0 nTitle=Background tracking nText=Enabled for Vehicle Tracking nIconLarge= nIconSmall= nIconColor= url=http://192.168.1.5:3500/service/gps/saveLocationForDriver?routeId=5ad08553a5172c2e88b2e8e3 syncUrl= syncThreshold=100 httpHeaders={authorization=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VySWQiOiI1YWQwODRmYWE1MTcyYzJlODhiMmU4ZTIiLCJyb2xlIjoiRFJJVkVSIiwiaWF0IjoxNTIzNjE1MDk0fQ.-vXyPccTqSCS0NTVT5iepCUfFfPcDaiNByuiqOjwL5k} maxLocations=10000 postTemplate={"bear":"@bearing","lon":"@longitude","lat":"@latitude"}]
04-13 18:38:25.075 29487 29487 D SystemWebChromeClient: file:///android_asset/www/js/angular/controllers/map-controller.js: Line 109 : is location enabled
04-13 18:38:25.075 29487 29487 I chromium: [INFO:CONSOLE(109)] "is location enabled", source: file:///android_asset/www/js/angular/controllers/map-controller.js (109)
04-13 18:38:25.076 29487   812 D com.marianhello.bgloc.BackgroundGeolocationFacade: Starting service
04-13 18:38:25.116 29487   812 I com.marianhello.bgloc.BackgroundGeolocationFacade: Starting bg service
04-13 18:38:25.134  3800  4078 D NetworkController.MobileSignalController(3): onSignalStrengthsChanged signalStrength=SignalStrength: 99 0 -120 -160 -120 -1 -1 15 -113 -9 68 2147483647 2147483647 2 2 gsm|lte level=2 voicelevel=2 datalevel=2
04-13 18:38:25.143 29487 29487 I com.marianhello.bgloc.LocationService: Creating LocationService
04-13 18:38:25.146  3800  4078 D NetworkController.MobileSignalController(4): onSignalStrengthsChanged signalStrength=SignalStrength: 22 0 -120 -160 -120 -1 -1 99 2147483647 2147483647 2147483647 2147483647 2147483647 4 4 gsm|lte level=4 voicelevel=4 datalevel=4
04-13 18:38:25.156 29487 29487 D AndroidRuntime: Shutting down VM
04-13 18:38:25.156 29487 29487 E AndroidRuntime: FATAL EXCEPTION: main
04-13 18:38:25.156 29487 29487 E AndroidRuntime: Process: com.followclass.driver, PID: 29487
04-13 18:38:25.156 29487 29487 E AndroidRuntime: java.lang.RuntimeException: Unable to create service com.marianhello.bgloc.LocationService: java.lang.SecurityException: uid 10571 cannot explicitly add accounts of type: com.followclass.driver.account
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:2951)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.app.ActivityThread.access$1900(ActivityThread.java:155)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1463)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:152)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5497)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: Caused by: java.lang.SecurityException: uid 10571 cannot explicitly add accounts of type: com.followclass.driver.account
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:1620)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:1573)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.accounts.IAccountManager$Stub$Proxy.addAccountExplicitly(IAccountManager.java:890)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.accounts.AccountManager.addAccountExplicitly(AccountManager.java:718)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at com.marianhello.bgloc.sync.AccountHelper.CreateSyncAccount(AccountHelper.java:23)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at com.marianhello.bgloc.LocationService.onCreate(LocationService.java:192)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:2941)
04-13 18:38:25.156 29487 29487 E AndroidRuntime: 	... 8 more
04-13 18:38:25.158 29487 29487 D AppTracker: App Event: crash
04-13 18:38:25.162  1124  4364 W ActivityManager:   Force finishing activity com.followclass.driver/.MainActivity
04-13 18:38:25.166  3800  3800 I StatusBarIconController:  appTransitionPending
04-13 18:38:25.168  7402  7402 W Settings: Setting airplane_mode_on has moved from android.provider.Settings.System to android.provider.Settings.Global, returning read-only value.
04-13 18:38:25.178  1124  4364 I ActivityManager_Thumbnails: Setting thumbnail of ActivityRecord{1832d97 u0 com.followclass.driver/.MainActivity t2846 f} to android.graphics.Bitmap@d2e3f00
04-13 18:38:25.183  1124   834 D DropBoxManagerService: file :: /data/system/dropbox/data_app_crash@1523624905172.txt
04-13 18:38:25.191  1124  1152 W ActivityManager:  Dismiss app error dialog : com.followclass.driver
04-13 18:38:25.191  1124  1152 W AppOps  : Finishing op nesting under-run: uid 1000 pkg android code 24 time=1464854877792 duration=9102 nesting=0
04-13 18:38:25.194 29487 29487 I Process : Sending signal. PID: 29487 SIG: 9
04-13 18:38:25.195  3800  3800 D PhoneStatusBar: setSystemUiVisibility vis=8000 mask=ffffffff oldVal=8600 newVal=8000 diff=600
04-13 18:38:25.222  1124  4362 I ActivityManager: Process com.followclass.driver (pid 29487) has died
04-13 18:38:25.222  1124  4294 I WindowState: WIN DEATH: Window{b7d2a6f u0 com.followclass.driver/com.followclass.driver.MainActivity}
04-13 18:38:25.222  1124  4362 D ActivityManager: cleanUpApplicationRecord -- 29487
04-13 18:38:25.222  1124  3831 D ConnectivityService: ConnectivityService NetworkRequestInfo binderDied(NetworkRequest [ id=360, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ], android.os.BinderProxy@dbc2e56)
04-13 18:38:25.223  1124  4390 D GraphicsStats: Buffer count: 8
04-13 18:38:25.223  1124  4362 W ActivityManager: Scheduling restart of crashed service com.followclass.driver/com.marianhello.bgloc.LocationService in 1000ms
04-13 18:38:25.223  1124  1138 D ConnectivityService: ConnectivityService NetworkRequestInfo binderDied(NetworkRequest [ id=359, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ], android.os.BinderProxy@bc21ed7)
04-13 18:38:25.223  1124  3724 D ConnectivityService: releasing NetworkRequest NetworkRequest [ id=360, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
04-13 18:38:25.223  1124  1150 V ActivityManager: killProcessGroupAsync took 0 ms for PID 29487 on thread 14
04-13 18:38:25.224  1124  3724 D ConnectivityService: sending notification RELEASED for NetworkRequest [ id=360, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
04-13 18:38:25.224  1124  3724 E ConnectivityService: RemoteException caught trying to send a callback msg for NetworkRequest [ id=360, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
04-13 18:38:25.224  1124  3724 D ConnectivityService: releasing NetworkRequest NetworkRequest [ id=359, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
04-13 18:38:25.225  1124  3724 D ConnectivityService: sending notification RELEASED for NetworkRequest [ id=359, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
04-13 18:38:25.225  1124  3724 E ConnectivityService: RemoteException caught trying to send a callback msg for NetworkRequest [ id=359, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
04-13 18:38:25.227  3800  3800 I StatusBarIconController:  appTransitionPending

**Comment by mauron85** [comment]
Your problem seems to be: "Unable to create service com.marianhello.bgloc.LocationService: java.lang.SecurityException: uid 10571 cannot explicitly add accounts of type: com.followclass.driver.account"

Not sure what it is. Maybe some "speciality" from Oneplus

**Comment by onkar0777** [comment]
Not sure if it is specific to Oneplus. I hoped you would have some insights. We have a working app using your plugin (many thanks for it 👍)  - https://play.google.com/store/apps/details?id=com.followclass.driver and it works great on my oneplus and every other device I have tried.

I used the 2.2.5 version of the plugin but I don't know what exact configuration was there of the other components.

**Comment by mauron85** [comment]
Glad to see real app that is using this plugin in production.

According https://stackoverflow.com/questions/33270613/android-app-crashes-on-addaccountexplicitlyaccount-password-null It seems like app id doesn't match sync account name.

I'll post instruction for you to check in next comment.

**Comment by mauron85** [comment]
So check if in `platforms/android/AndroidManifest.xml` `package` matches `content_authority` in `platforms/android/res/values/strings.xml`.

**Comment by The-Don-Himself** [comment]
By the way, I was getting this a lot as well during development, I never got the time to dig too much into it but changing the provider from ACTIVITY_PROVIDER to DISTANCE_FILTER_PROVIDER solved it for me.

@onkar0777 could you try that and see if it solves your issue, if it does perhaps this info can help @mauron85 to pinpoint the exact cause.

**Comment by onkar0777** [comment]
The above values match but the issue seems to do something in the package name. I did not mention a hack I did to get the app running, which seems to be the root cause of this. So there are 3 cases:

1. The hack i did - I changed the variable `CONTENT_AUTHORITY` in `config.xml` to `$PACKAGE_NAME.test` and `android:authorities` in `plugins/cordova-plugin-mauron85-background-geolocation/plugin.xml` to "com.followclass.driver.test" --- **This made the app run but lead to the above errors and ultimately the app crashed.**

2. I remove and add the android platform, keep everything the same - use package name `com.followclass.driver` - I am unable to run my app on my device - `ionic run android` fails with this message - Failure [INSTALL_FAILED_CONFLICTING_PROVIDER]

3. I changed the id (package name) in `config.xml` to something else - `com.followclass.driver_test` - Remove and add android platform - Everything works fine - S

So I think the actual issue is point 2 where if I use my correct package name - com.followclass.driver it leads to app not being installed on the device.

Should I open a new issue for this or is it ok to continue here ?

**Comment by mauron85** [comment]
According https://stackoverflow.com/a/16267987/3896616

>The authority, as listed in android:authorities must be unique.

Meaning you should be fine if you uninstall conflicting app first (completely for all users in android settings)

**Comment by onkar0777** [comment]
I had tried this multiple times earlier (even with adb uninstall com.followclass.driver) but it did not work. Now in a desperate attempt I uninstall all the apps created by me and restarted the device.

Now it seems to be working. I will update to v3 completely now and hope everything will go on smoothly. Thank you very much for your support and prompt replies. Much appreciated 👍 
Closing the issue.
